toqito/helper/bell_notation_conversions.py

```diff
@@ -0,0 +1,748 @@
+    >>> p_fc = np.array([[1, 0, 0], [0, 0, 0], [0, 0, 0]])
+    >>> fc_to_cg(p_fc, behaviour=True)
+    array([[1.  , 0.5 , 0.5 ],
+           [0.5 , 0.25, 0.25],
+           [0.5 , 0.25, 0.25]])
+
+    :param fc_mat: The matrix in Full Correlator notation.
+    :param behaviour: If True, assume input is a behaviour (default: False, assume functional).
+    :return: The matrix in Collins-Gisin notation.
+
+    .. note::
+        This function is adapted from the QETLAB MATLAB package function `FC2CG`.
+
+    References
+    ==========
+    .. bibliography::
+        :filter: docname in docnames
+
+    """
+    ia = fc_mat.shape[0] - 1
+    ib = fc_mat.shape[1] - 1
+
+    cg_mat = np.zeros((ia + 1, ib + 1))
+
+    a_vec = fc_mat[1:, 0]
+    b_vec = fc_mat[0, 1:]
+    c_mat = fc_mat[1:, 1:]
+
+    if not behaviour:
+        cg_mat[0, 0] = fc_mat[0, 0] + np.sum(c_mat) - np.sum(a_vec) - np.sum(b_vec)
+        cg_mat[1:, 0] = 2 * a_vec - 2 * np.sum(c_mat, axis=1)
+        cg_mat[0, 1:] = 2 * b_vec - 2 * np.sum(c_mat, axis=0)
+        cg_mat[1:, 1:] = 4 * c_mat
+    else:
+        cg_mat[0, 0] = 1
+        cg_mat[1:, 0] = (1 + a_vec) / 2
+        cg_mat[0, 1:] = (1 + b_vec) / 2
+        cg_mat[1:, 1:] = (np.ones((ia, ib)) + a_vec[:, np.newaxis] + b_vec[np.newaxis, :] + c_mat) / 4
+
+    return cg_mat
+
+
+def cg_to_fp(cg_mat: np.ndarray, desc: list[int], behaviour: bool = False) -> np.ndarray:
+    r"""Convert a Bell functional or behaviour from Collins-Gisin (CG) to Full Probability (FP) notation.
+
+    The Collins-Gisin (CG) notation for a Bell functional or behaviour is represented by a matrix
+    (see :func:`cg_to_fc`). The Full Probability (FP) notation represents the full probability
+    distribution :math:`V(a, b, x, y) = P(a, b | x, y)`, the probability of Alice getting outcome
+    :math:`a` (0 to oa-1) and Bob getting outcome :math:`b` (0 to ob-1) given inputs :math:`x`
+    (0 to ia-1) and :math:`y` (0 to ib-1). This is stored as a 4D numpy array with indices
+    `V[a, b, x, y]`.
+
+    This function converts from CG to FP notation.
+
+    Examples
+    ==========
+
+    Consider the CHSH inequality functional in CG notation:
+
+    .. math::
+        \text{CHSH}_{CG} =
+        \begin{pmatrix}
+            0 & 0 & 0 \\
+            0 & 1 & -1 \\
+            0 & -1 & 1
+        \end{pmatrix}
+
+    Converting to FP notation (desc = [2, 2, 2, 2]):
+
+    >>> import numpy as np
+    >>> from toqito.helper import cg_to_fp
+    >>> chsh_cg = np.array([[0, 0, 0], [0, 1, -1], [0, -1, 1]])
+    >>> desc = [2, 2, 2, 2] # oa, ob, ia, ib
+    >>> cg_to_fp(chsh_cg, desc) # doctest: +NORMALIZE_WHITESPACE
+    array([[[[ 1., -1.],
+             [-1.,  1.]],
+            [[ 0.,  0.],
+             [ 0.,  0.]]],
+           [[[ 0.,  0.],
+             [ 0.,  0.]],
+            [[ 0.,  0.],
+             [ 0.,  0.]]]])
+
+
+    Consider a behaviour (probability distribution) in CG notation (desc = [2, 2, 2, 2]):
```

Comment:
I think @purva-thakre has mentioned this in another comment, and my 🇺🇸 -ness is showing, but `behaviour` -> `behavior` :)